gymnasium/envs/mujoco/humanoid_v5.py

```diff
@@ -0,0 +1,516 @@
+    version > 2.0).
+
+
+    ## Rewards
+    The reward consists of three parts:
+    - *healthy_reward*:
+    Every timestep that the Humanoid is alive (see section Episode Termination for definition),
+    it gets a reward of fixed value `healthy_reward`.
+    - *forward_reward*:
+    A reward for moving forward,
+    this reward would be positive if the Humanoid moves forward (in the positive $x$ direction / in the right direction).
+    $w_{forward} \times \frac{dx}{dt}$, where
+    $dx$ is the displacement of the center of mass ($x_{after-action} - x_{before-action}$),
+    $dt$ is the time between actions, which depends on the `frame_skip` parameter (default is 5),
+    and `frametime` which is 0.001 - so the default is $dt = 5 \times 0.003 = 0.015$,
+    $w_{forward}$ is the `forward_reward_weight` (default is $1.25$).
+    The center of mass calculation is defined in the `.py` file for the Humanoid.
+    - *ctrl_cost*:
+    A negative reward to penalize the Humanoid for taking actions that are too large.
+    $w_{control} \times \\|action\\|_2^2$,
+    where $w_{control}$ is `ctrl_cost_weight` (default is $0.1$).
+    If there are *nu* actuators/controls, then the control has shape `nu x 1`.
+    - *contact_cost*:
+    A negative reward to penalize the Humanoid if the external contact forces are too large.
+    $w_{contact} \times clamp(contact\\_cost\\_range, \\|F_{contact}\\|_2^2)$, where
+    $w_{contact}$ is `contact_cost_weight` (default is $5\times10^{-7}$),
+    $F_{contact}$ are the external contact forces (see `cfrc_ext` section on observation).
+
+    The total reward returned is ***reward*** *=* *healthy_reward + forward_reward - ctrl_cost - contact_cost*
+    and `info` will also contain the individual reward terms.
+
+    Note: in `v4` the total reward returned is ***reward*** *=* *healthy_reward + forward_reward - ctrl_cost*
+
+
+    ## Starting State
+    All observations start in state
+    (0.0, 0.0,  1.4, 1.0, 0.0  ... 0.0) with a uniform noise in the range
+    of [-`reset_noise_scale`, `reset_noise_scale`] added to the positional and velocity values (values in the table)
+    for stochasticity. Note that the initial z coordinate is intentionally
+    selected to be high, thereby indicating a standing up humanoid. The initial
+    orientation is designed to make it face forward as well.
+
+
+    ## Episode End
+    #### Termination
+    If `terminate_when_unhealthy is True` (which is the default), the environment terminates when the Humanoid is unhealthy.
+    The Humanoid is said to be unhealthy if any of the following happens:
+
+    1. The z-position of the torso (the height) is no longer contained in `healthy_z_range`.
+
+    #### Truncation
+    The maximum duration of an episode is 1000 timesteps.
+
+
+    ## Arguments
+    `gymnasium.make` takes additional arguments such as `xml_file`, `ctrl_cost_weight`, `reset_noise_scale`, etc.
```

Comment:
Humanoid provides a range of parameters to modify the observation space, reward function, and termination condition. These parameters can be applied during `gymnasium.make` in the following way